### ISSUE TITLE
[NotifyAll] - Don't run into "504 Gateway Time-out"

### ISSUE THREAD

**Issue by nupplaphil** [friendica/friendica#7220]
### Expected behavior

Send an e-mail to ~100 users without an error.

### Actual behavior

After a while, I get a `504 Gateway Time-out

I know that I can expand the timeout, but I think it would be better to add a "background worker" for such amount of mails instead of a frontend execution. Therefore I set it as an enhancement.

### Steps to reproduce the problem

- Add ~100 user to your node
- send a "notify all" to all of them

### Friendica version you encountered the problem

2019.06-rc

### Friendica source (git, zip)

git

**Comment by annando** [comment]
This has to be solved by creating worker jobs for that task. So either could the whole mail loop be a single worker task or we create a task for every receiver.

**Comment by nupplaphil** [comment]
As you're creating most of the logic, what do you think is better? I'd be for one per receiver due to better scalability, but than the creation of each worker has to be fast as well.. So I'm not sure..

**Comment by MrPetovan** [comment]
Middle of the road is creating a worker task for each 10/50/100 recipients.

**Comment by annando** [comment]
Creating a single task per receiver is no big deal. We always are doing so when transmitting our messages. And when send with medium or low priority, then there shouldn't be that much parallel mail tasks.

**Comment by ben-utzer** [comment]
Yes, please lets enhance this addon, it would be great to be able to write a message to all users of a server.
